_posts/2018-01-05-python101.md

```diff
@@ -0,0 +1,847 @@
+
+  def get_overall_grade_for(self, student):
+    sum_scores = 0  # need an accumulator (see Glossary) to compute average
+    num_graded = 0  # need to keep track of total number of grades for average
+    for assignment, grades in self.assignments.items(): # iterate over all items in self.assignments
+      if student in grades.keys():  # look to see if the student has a grade
+        sum_scores += grades[student]  # add that grade to the accumulator
+        num_graded += 1 # increment number of graded assignments (the denominator)
+
+    if num_graded == 0: # can't have an average with 0 elements
+      return "N/A"
+
+    return sum_scores / float(num_graded) # return the average
+```
+
+So how do we use our classes now? Well, we create some instances of the classes and call the functions! We do that in the main section of our program.
+
+```python
+john = Student("John", "A", "Smith") # create a Student named John
+katie = Student("Katie", "M", "Zhu") # create a Student named Katie
+
+biology = Course("Biology") # create a course called Biology
+calculus = Course("Calculus") # create a course called Calculus
+physics = Course("Physics") # create a course called Physics
+
+john.enroll(biology) # enroll John in Bio and Calculus
+john.enroll(calculus)
+
+katie.enroll(biology) # enroll Katie in Bio, Calc, and Physics
+katie.enroll(calculus)
+katie.enroll(physics)
+
+biology.add_assignment("Bio exam #1") # Add the assignments for each of the three classes
+biology.add_assignment("Bio exam #2")
+physics.add_assignment("Physics homework #1")
+calculus.add_assignment("Calc homework #1")
+calculus.add_assignment("Calc quiz #1")
+
+biology.set_grade("Bio exam #1", john, 90) # give John and Katie grades for Bio
+biology.set_grade("Bio exam #1", katie, 95)
+
+physics.set_grade("Physics homework #1", katie, 86) # give Katie a grade for Physics
+
+calculus.set_grade("Calc homework #1", john, 100) # give John and Katie grades for Calculus
+calculus.set_grade("Calc homework #1", katie, 98)
+calculus.set_grade("Calc quiz #1", john, 78)
+calculus.set_grade("Calc quiz #1", katie, 84)
+
+john.get_transcript() # print out their overall grades
+katie.get_transcript()
+```
+Output:
+```
+John A. Smith:
+	Biology: 90.0
+	Calculus: 89.0
+Katie M. Zhu:
+	Biology: 95.0
+	Calculus: 91.0
+	Physics: 86.0
+```
+
+# <a id="conclusion">Conclusion</a>
+This is by no means a very extensive tutorial but we hope that you will feel more confident about pursuing coding projects now.
+
+Below are some other handy resources for you to learn more about Python!
+* [https://python101.pythonlibrary.org/intro.html](https://python101.pythonlibrary.org/intro.html)
+* [https://www.codecademy.com/learn/learn-python](https://www.codecademy.com/learn/learn-python)
+
+# <a id="glossary">Glossary</a>
+
+<dl>
+  <lt>class</lt>
+  <lt>double</lt>
+  <lt>variable</lt>
+  <lt>integer</lt>
+  <lt>float</lt>
+  <lt>object</lt>
+  <lt>instance</lt>
+</dl>
```

Comment:
is the glossary just supposed to be a list of words like this or are you going to hyperlink as well?